serverless/aws/features/stepfunctions.py

```diff
@@ -337,8 +337,15 @@ def __init__(
         self.definition = Definition(description, auto_fallback, auto_catch)
         self.events = events or []
 
-    def task(self, function, end: Optional[bool] = None):
-        return self.definition.add(Task(function, end=end))
+    
+    def task(self, function=None, resource=None, name=None, end: Optional[bool] = None):
```

Comment:
Wouldn't this code behave this same way: 
```python
    def task(self, function=None, resource=None, name=None, end: Optional[bool] = None):
        return self.definition.add(Task(function=function, resource=resource, name=name, end=end))
```

Reply:
This allows step functions to add services that are created outside the current serverless.yml. Earlier, task function was only taking the function as the argument. With this change, a new Task can be created with either function or a resource. Task was already taking resource but task function in stepfunctions.py was not exposing the resources. 

Usage is in PR https://github.com/epsyhealth/gdpr-user-data-extraction-service/blob/2cd5e25f81d4f278755005b12eeb14305180bed5/serverless.yml.py#L97-L107

Reply:
What I am saying is that: 
```python
def task(self, function=None, resource=None, name=None, end: Optional[bool] = None):
        if function is not None and resource is None:
            return self.definition.add(Task(function, end=end))
        elif function is None and resource is not None:
            return self.definition.add(Task(resource=resource, name=name, end=end))
        else:
            raise ValueError("Either 'function' or 'resource' must be provided, but not both.")
```

and

```python
    def task(self, function=None, resource=None, name=None, end: Optional[bool] = None):
        return self.definition.add(Task(function=function, resource=resource, name=name, end=end))
```

produces same output. 

Or I missed something?

Reply:
I see your question now. sorry, i didnt understand before. :-)
Yes, they are the same but as this is a public repo/tool that could be used by others. So, I wanted the usage (combination of right args) to be more intuitive/clear to the end user. 
Thats all.

Reply:
@mg-epsy how is that cleaner? Main question is, is your six lines doing same as one line from my snippet?

Reply:
ok, I will change to the one liner.